octobot_trading/exchanges/implementations/future_ccxt_exchange.py

```diff
@@ -167,3 +171,58 @@ def clean_trade(self, trade):
 
     def clean_order(self, order):
         return self.connector.clean_order(order)
+
+    async def get_open_positions(self, **kwargs: dict) -> list:
+        return await self.connector.fetch_positions()
+
+    async def get_funding_rate(self, symbol: str, **kwargs: dict) -> dict:
+        return await self.connector.fetch_funding_rate(symbol=symbol)
+
+    async def get_funding_rate_history(self, symbol: str, limit: int = 1, **kwargs: dict) -> list:
+        return await self.connector.fetch_funding_rate_history(symbol=symbol, limit=limit)
+
+    async def set_symbol_leverage(self, symbol: str, leverage: int):
+        return await self.connector.set_leverage(leverage=leverage, symbol=symbol)
+
+    async def set_symbol_margin_type(self, symbol: str, isolated: bool):
+        return await self.connector.set_margin_mode(symbol=symbol,
+                                                    marginType=self.CCXT_ISOLATED if isolated else self.CCXT_CROSSED)
+
+    def parse_position(self, position_dict) -> dict:
+        try:
+            position_dict.update({
+                trading_enums.ExchangeConstantsPositionColumns.SYMBOL.value:
+                    position_dict.get(position_dict[trading_enums.ExchangePositionCCXTColumns.SYMBOL.value]),
+                trading_enums.ExchangeConstantsPositionColumns.ID.value:
+                    position_dict.get(position_dict[trading_enums.ExchangePositionCCXTColumns.SYMBOL.value]),
+                trading_enums.ExchangeConstantsPositionColumns.QUANTITY.value:
+                    position_dict.get(trading_enums.ExchangePositionCCXTColumns.CONTRACT_SIZE.value, constants.ZERO),
```

Comment:
no need to cast it into decimal.Decimal here ?

Reply:
Yes we need a cast